src/app/user/settings/page.tsx

```diff
@@ -0,0 +1,232 @@
+"use client";
+
+import Card from "@/components/Card";
+import { Title } from "@/components/Title";
+import FormField from "@/components/FormField";
+import { Button } from "@/components/ui/button";
+import { Alert, AlertTitle } from "@/components/ui/alert";
+import { useState, useEffect } from "react";
+import { useRouter } from "next/navigation";
+import { useAuth } from "@/app/AuthProvider";
+import { auth } from "@/lib/firebase";
+import {
+  updateEmail as fbUpdateEmail,
+  updatePassword as fbUpdatePassword,
+  reauthenticateWithCredential,
+  EmailAuthProvider,
+  deleteUser,
+} from "firebase/auth";
+
+type Tab = "email" | "password" | "delete";
+
+export default function Setting() {
+  const { user, logout, loading } = useAuth();
+  const router = useRouter();
+
+  const [tab, setTab] = useState<Tab>("email");
+  const [error, setError] = useState("");
+  const [success, setSuccess] = useState("");
+
+  const currentEmail = user?.email ?? "";
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newEmail, setNewEmail] = useState(currentEmail);
+  const [newPassword, setNewPassword] = useState("");
+
+  const resetAlerts = () => {
+    setError("");
+    setSuccess("");
+  };
+
+  useEffect(() => {
+    if (!loading && !user) {
+      router.replace("/user/login");
+    }
+  }, [loading, user, router]);
+
+  if (loading || !user) return null;
+
+  async function handleEmailUpdate(e: React.FormEvent) {
```

Comment:
I wasn't able to update email address since I need to validate my email first. However, I did not receive and email or similar to validation my email.